File: utbot-framework/src/main/java/org/utbot/quickcheck/generator/java/util/function/ToLongFunctionGenerator.java


package org.utbot.quickcheck.generator.java.util.function;
import org.utbot.engine.greyboxfuzzer.util.UtModelGenerator;
import org.utbot.framework.plugin.api.UtModel;

import org.utbot.quickcheck.generator.ComponentizedGenerator;
import org.utbot.quickcheck.generator.GenerationStatus;
import org.utbot.quickcheck.generator.Generator;
import org.utbot.quickcheck.generator.Generators;
import org.utbot.quickcheck.random.SourceOfRandomness;

import java.util.function.ToLongFunction;

import static org.utbot.external.api.UtModelFactoryKt.classIdForType;
import static org.utbot.quickcheck.generator.Lambdas.makeLambda;

/**
 * Produces values of type {@link ToLongFunction}.
 *
 * @param <T> type of parameter of produced function
 */
public class ToLongFunctionGenerator<T>
    extends ComponentizedGenerator<ToLongFunction> {

    private Generator<Long> generator;

    public ToLongFunctionGenerator() {
        super(ToLongFunction.class);
    }

    @Override
    public void provide(Generators provided) {
        super.provide(provided);

        generator = gen().type(long.class);
    }

    @Override public UtModel generate(
        SourceOfRandomness random,
        GenerationStatus status) {

        return UtModelGenerator.getUtModelConstructor().construct(makeLambda(ToLongFunction.class, generator, status), classIdForType(ToLongFunction.class));
    }

    @Override public int numberOfNeededComponents() {
        return 1;
    }
}

File: utbot-framework/src/main/kotlin/org/utbot/quickcheck/generator/java/util/function/ToLongFunctionGenerator.kt
package org.utbot.quickcheck.generator.java.util.function

import org.utbot.engine.greyboxfuzzer.util.UtModelGenerator.utModelConstructor
import org.utbot.external.api.classIdForType
import org.utbot.framework.plugin.api.UtModel
import org.utbot.quickcheck.generator.ComponentizedGenerator
import org.utbot.quickcheck.generator.GenerationStatus
import org.utbot.quickcheck.generator.Generator
import org.utbot.quickcheck.generator.Generators
import org.utbot.quickcheck.generator.Lambdas.Companion.makeLambda
import org.utbot.quickcheck.random.SourceOfRandomness
import java.util.function.ToLongFunction

/**
 * Produces values of type [ToLongFunction].
 *
 * @param <T> type of parameter of produced function
</T> */
class ToLongFunctionGenerator<T> : ComponentizedGenerator(ToLongFunction::class.java) {
    private var generator: Generator? = null

    override fun createModifiedUtModel(random: SourceOfRandomness, status: GenerationStatus): UtModel {
        return generate(random, status)
    }

    override fun provide(provided: Generators) {
        super.provide(provided)
        generator = gen()!!.type(Long::class.javaPrimitiveType!!)
    }

    override fun generate(
        random: SourceOfRandomness,
        status: GenerationStatus
    ): UtModel {
        return utModelConstructor.construct(
            makeLambda(
                ToLongFunction::class.java, generator!!, status
            ), classIdForType(ToLongFunction::class.java)
        )
    }

    override fun numberOfNeededComponents(): Int {
        return 1
    }
}